### PR TITLE
Performance work

### LINKED ISSUE #283: diffexp requests (REST 0.2) do not consolidate OBS index selector with ranges
potential performance improvement: use range selectors for index filters when hitting /diffexp/obs

currently, every obs is individually specified.   For many selection sets, using range selectors will significantly reduce the size of the request payload.

eg, change `[0, 1, 2, 3]` to `[ [0,3] ]`

### PR DESCRIPTION
**Pull request by bkmartinjr**
Changes:
* Range encode index filters in requests to /diffexp - cuts down on request payload size, in some cases dramatically
* upgrade to latest version of scanpy/anndata (required by next bullet)
* enable cached hd5a file open, reducing startup time for data
* during initialization of CLI, call layout with parameters that are more likely to match front-end (ie, don't waste time memoizing something we won't use)

Fixes #283

### CODE REVIEW

File: server/app/scanpy_engine/scanpy_engine.py [hunk 1]
```diff
@@ -78,7 +78,10 @@ def add_to_parser(cls, subparsers, invocation_function):
 
     @staticmethod
     def _load_data(data):
-        return sc.read(os.path.join(data, "data.h5ad"))
+        # See https://scanpy.readthedocs.io/en/latest/api/scanpy.api.read.html
+        # Based upon this advice, setting both backend= and cache= parameters.
+        # return sc.read(os.path.join(data, "data.h5ad"), cache = True)
+        return sc.read(os.path.join(data, "data.h5ad"), backed='r', cache=True)
```
Comment (csweaver) [inline server/app/scanpy_engine/scanpy_engine.py]: I didn't quite understand the difference between `r` and `r+` options for `backed`. Does r+ mean if you wanted to save changes back?

**Comment by bkmartinjr** [comment]
@csweaver - you are fast.  I wasn't quite ready for review.  Please look at the change to the init-time call to layout. Turns out that this call was wasted as the init-time call arguments did not match the clients initial call - so the extra work caching was for naught.   This last change makes them match.

File: server/app/scanpy_engine/scanpy_engine.py [hunk 2]
```diff
@@ -327,6 +331,8 @@ def layout(self, filter, interactive_limit=None):
         :param interactive_limit: -- don't compute if total # genes in dataframes are larger than this
         :return:  [cellid, x, y, ...]
         """
+        print(filter)
+        print(interactive_limit)
```
Comment (csweaver) [inline server/app/scanpy_engine/scanpy_engine.py]: Remove print statements please

**Comment by csweaver** [comment]
@bkmartinjr the `None->{}` change looks good too. You can actually ping me again to look at changes by selecting me as a reviewer after I've approved or rejected changes.

**Pull request merged**